### PR TITLE
Support timm zero shot learning

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Use timm to create head to cover the case that, if the provided `num_classes` is the same as the pre-trained `num_classes`, we can reuse the pre-trained head.
2. Move the timm head outside since we need to get the pooled features for late fusion sometimes.
3. Output logits only if the `num_classes` > 0.
4. Add a unit test for timm zero shot learning.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/models/timm_image.py [hunk 1]
```diff
@@ -114,7 +115,9 @@ def forward(
         image_valid_num = batch[self.image_valid_num_key]
         ret = {COLUMN_FEATURES: {FEATURES: {}, MASKS: {}}}
         if self.mix_choice == "all_images":  # mix inputs
-            mixed_images = images.sum(dim=1) / image_valid_num[:, None, None, None]  # mixed shape: (b, 3, h, w)
+            mixed_images = (
```
Comment (bryanyzhu) [inline multimodal/src/autogluon/multimodal/models/timm_image.py:118]: I don't understand this part. Can you explain a bit? Thank you

File: multimodal/tests/unittests/test_zero_shot.py [hunk 2]
```diff
@@ -98,3 +103,36 @@ def test_clip_zero_shot():
     # invalid API usage 2: predicting probability with only one dictionary as input.
     with pytest.raises(AssertionError):
         prob = predictor.predict_proba({"image": [cat_image_name], "text": [cat_text]})
+
+
+def test_timm_zero_shot():
+    cat_image_name, dog_image_name = download_sample_images()
+
+    predictor = MultiModalPredictor(
+        hyperparameters={
```
Comment (bryanyzhu) [inline multimodal/tests/unittests/test_zero_shot.py:112]: Can we test some CNN models as well? (maybe at least a ResNet50) CNN has some structural differences than transformers, that may break current code logic.

**Review by bryanyzhu — CHANGES_REQUESTED** [review]
LGTM, left some comments. It would be great to add some unittest for CNN models in TIMM.

File: multimodal/src/autogluon/multimodal/models/timm_image.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/models/timm_image.py:118]: Each row in a dataframe may have multiple images. `all_images` means mixing the images into one before the model forward. If no valid images exist in a row, the `image_valid_num` is 0. To avoid the divide-by-zero error, we clamp the min value to 1e-6.

File: multimodal/tests/unittests/test_zero_shot.py (hunk 2 above)
Comment (zhiqiangdon) [reply multimodal/tests/unittests/test_zero_shot.py:121]: Thanks. Added CNN tests.

File: multimodal/src/autogluon/multimodal/models/timm_image.py (hunk 1 above)
Comment (bryanyzhu) [reply multimodal/src/autogluon/multimodal/models/timm_image.py:118]: Thanks

**Review by bryanyzhu — APPROVED** [review]
